source/custom/nidcpower_service.custom.cpp

```diff
@@ -43,16 +44,33 @@ ::grpc::Status NiDCPowerService::MeasureMultiple(::grpc::ServerContext* context,
       ViReal64* current_measurements = response->mutable_current_measurements()->mutable_data();
 
       auto status = library_->MeasureMultiple(vi, channel_name, voltage_measurements, current_measurements);
+      if (status < VI_SUCCESS) {
```

Comment:
Can you do this check instead of the CheckStatus on line 40? The CheckStatus will throw a `DriverErrorException` on warnings which will now cause the UNKNOWN gRPC status.

Or you could also change CheckStatus to do this check instead of `status != 0` and then wrap both library_ calls in it.

Reply:
I definitely don't want a warning to return UNKNOWN. I need to evaluate each of these more closely.